Content.Server/GameObjects/Components/AdvertiseComponent.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using Content.Server.Advertisements;
+using Content.Server.Interfaces.Chat;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Log;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+namespace Content.Server.GameObjects.Components
+{
+    [RegisterComponent]
+    public class AdvertiseComponent : Component
+    {
+        [Dependency] private readonly IChatManager _chatManager = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
```

Comment:
IChatManager + IRobustRandom can probably just be resolved directly given it only fires off every 40-80 seconds by default
IPrototypeManager is used once so just resolve it locally so you don't keep it around in memory

Normally we only store the member ones if they're being used frequently (and even then if you use an ECS approach you can just store them on a system instead).

Reply:
Fixed